pkg/host-disk/host-disk.go

```diff
@@ -88,18 +88,27 @@ func dirBytesAvailable(path string) (uint64, error) {
 	return stat.Bavail * uint64(stat.Bsize), nil
 }
 
-func createSparseRaw(fullPath string, size int64) error {
+func createSparseRaw(fullPath string, size int64) (err error) {
+	var f *os.File
 	offset := size - 1
-	f, err := os.Create(fullPath)
-	defer f.Close()
+	f, err = os.Create(fullPath)
+
 	if err != nil {
-		return err
+		return
 	}
+
+	defer func() {
```

Comment:
How about using a wrapper to do this?
`defer safeClose(f.close, &err)`

Reply:
We can use a wrapper as I did in libvirt_helper.go  , I personally think that a wrapper is more reasonable where there are multiple calls and for such single call I think inline function is fine. 
I will change to wrapper.